### PR TITLE
textarea and submit button created. API request fails

### PR DESCRIPTION
**Pull request by txu89**
Getting error when trying to hit API
`Failed to load http://localhost:4567/deal: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://localhost:3000' is therefore not allowed access.`

### CODE REVIEW

**Comment by teddylo57** [comment]
https://stackoverflow.com/questions/18642828/origin-http-localhost3000-is-not-allowed-by-access-control-allow-origin

**Comment by teddylo57** [comment]
Seems to be a CORS issue and we need to add a header (since 3000 port vs 4567)

File: frontend/src/components/Textarea.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import React, { Component } from 'react';
+
+export class SubmitHand extends Component {
```
Comment (ericliu121187) [inline frontend/src/components/Textarea.js]: Usually, the exported class name of the component is the same name as the file name and I'd prefer components to be more noun-y.

Maybe `HandSubmission`?

File: frontend/src/components/App.js [hunk 2]
```diff
@@ -1,4 +1,6 @@
 import React, { Component } from 'react';
+import { SubmitHand } from './Textarea.js';
```
Comment (ericliu121187) [inline frontend/src/components/App.js]: No need to include `.js`

File: frontend/src/components/Textarea.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+  handleUserInput(e) {
+    this.setState({
+      userInput: e.target.value
+    })
+  }
+
+  getDeal() {
+
+    function httpGetAsync(Url, callback) {
```
Comment (ericliu121187) [inline frontend/src/components/Textarea.js]: Let's use the native Fetch API for our http requests.

https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch

File: frontend/src/components/Textarea.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+      xmlHttp.send(null)
+    }
+
+    httpGetAsync('http://localhost:4567/deal', function(response) {
+        console.log(response)
+      })
+  }
+
+  postHand() {
+    this.setState({
```
Comment (ericliu121187) [inline frontend/src/components/Textarea.js]: though I suspect this is a WIP, just pointing out this is kinda odd. why set the state of hand to the state of userInput here?

File: frontend/src/components/Textarea.js (hunk 1 above)
Comment (txu89) [reply frontend/src/components/Textarea.js]: Changed to HandSubmission

File: frontend/src/components/App.js (hunk 2 above)
Comment (txu89) [reply frontend/src/components/App.js]: removed `.js`

File: frontend/src/components/Textarea.js (hunk 3 above)
Comment (txu89) [reply frontend/src/components/Textarea.js]: Swtiched to Fetch instead of XHR

File: frontend/src/components/Textarea.js (hunk 4 above)
Comment (txu89) [reply frontend/src/components/Textarea.js]: This was just testing from earlier to see if button was working before I was able to do a GET request

**Comment by ericliu121187** [comment]
kk, let's get this cleaned up and merge this in -- i'l create a new ticket for myself to check out the bug 

this is close -- fix the setState mentioned above and clean up the syntax and formatting, thanks

File: frontend/src/components/Textarea.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  }
+
+  handleUserInput(e) {
+    this.setState({
+      userInput: e.target.value
+    });
+  }
+
+  postHand() {
+    const input = {
```
Comment (ericliu121187) [inline frontend/src/components/Textarea.js]: change name maybe data or body not input, input means a shitload of other things in FE world and also body/data ppl are more used to

also remove the double quotes from the keys in this object -- this isn't a JSON object, it's a JS object

usually also we use single-quotes for the string values, but nbd right now -- i'll set up a linter later on

File: frontend/src/components/Textarea.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+      "handHistoryId": 1908745,
+      "hero": {},
+      "players": [],
+      "preflop": {},
+      "river": {},
+      "turn": {}
+    };
+
+
+    const handsUrl = 'http://localhost:4567/hands';
```
Comment (ericliu121187) [inline frontend/src/components/Textarea.js]: remove extra line break above

this can be `HANDS_URL`

File: frontend/src/components/Textarea.js [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+      'Accept': 'application/json, text/plain, */*',
+      'Content-Type': 'application/json',
+      'X-Origin': '*'
+    };
+
+
+    // let input = JSON.stringify(JSON.parse(this.state.userInput));
+    let myHeaders = new Headers(httpHeaders);
+
+    let fetchData = {
```
Comment (ericliu121187) [inline frontend/src/components/Textarea.js]: this should be a const and be named as `fetchOptions` not data

File: frontend/src/components/Textarea.js [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+
+    const httpHeaders = {
+      'Accept': 'application/json, text/plain, */*',
+      'Content-Type': 'application/json',
+      'X-Origin': '*'
+    };
+
+
+    // let input = JSON.stringify(JSON.parse(this.state.userInput));
+    let myHeaders = new Headers(httpHeaders);
```
Comment (ericliu121187) [inline frontend/src/components/Textarea.js]: const me up baby

File: frontend/src/components/Textarea.js [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+        headers: myHeaders,
+        method: 'POST',
+        mode: 'cors'
+
+    };
+
+    fetch(handsUrl, fetchData
+    ).then(res => res.json()
+    ).then(res => console.log(res)
+    );
```
Comment (ericliu121187) [inline frontend/src/components/Textarea.js]: maybe fix up the indentation here?

File: frontend/src/components/Textarea.js [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+
+  }
+
+  render() {
+    return (
+      <div>
+        <textarea rows = "10" cols="50"
```
Comment (ericliu121187) [inline frontend/src/components/Textarea.js]: remove spaces between rows = "10"

File: frontend/src/components/Textarea.js (hunk 5 above)
Comment (ericliu121187) [reply frontend/src/components/Textarea.js]: oh nvm about renaming it -- input works i guess it makes sense in this case as it is representing the user input

File: frontend/src/components/Textarea.js [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+
+    let fetchData = {
+        body: JSON.stringify(input),
+        headers: myHeaders,
+        method: 'POST',
+        mode: 'cors'
+
+    };
+
+    fetch(handsUrl, fetchData
```
Comment (ericliu121187) [inline frontend/src/components/Textarea.js]: Can you throw in a comment saying like // TODO fix this, currently doesn't work

Comment (txu89) [reply frontend/src/components/Textarea.js]: Added

File: frontend/src/components/Textarea.js [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+      'Accept': 'application/json, text/plain, */*',
+      'Content-Type': 'application/json',
+      'X-Origin': '*'
+    };
+
+
+    // let input = JSON.stringify(JSON.parse(this.state.userInput));
+    const myHeaders = new Headers(httpHeaders);
+
+    const fetchData = {
```
Comment (ericliu121187) [inline frontend/src/components/Textarea.js]: `fetchOptions` not `fetchData`

File: frontend/src/components/Textarea.js [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+        headers: myHeaders,
+        method: 'POST',
+        mode: 'cors'
+
+    };
+
+     //TODO: Fix this, /POST not working
+    fetch(handsUrl, fetchData
+    ).then(res => res.json()
+    ).then(res => console.log(res)
```
Comment (ericliu121187) [inline frontend/src/components/Textarea.js]: indent this better?

```javascript
fetch(handsUrl, fetchData)
  .then(res => res.json())
  .then(res => console.log(res));
```

File: frontend/src/components/Textarea.js (hunk 5 above)
Comment (txu89) [reply frontend/src/components/Textarea.js]: Left as input. Double quotes removed.

File: frontend/src/components/Textarea.js (hunk 6 above)
Comment (txu89) [reply frontend/src/components/Textarea.js]: Changed

File: frontend/src/components/Textarea.js (hunk 10 above)
Comment (txu89) [reply frontend/src/components/Textarea.js]: Spaces removed

File: frontend/src/components/Textarea.js (hunk 12 above)
Comment (txu89) [reply frontend/src/components/Textarea.js]: Changed

File: frontend/src/components/Textarea.js (hunk 13 above)
Comment (txu89) [reply frontend/src/components/Textarea.js]: Changed

File: frontend/src/components/Textarea.js [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+    const HANDS_URL = 'http://localhost:4567/hands';
+
+    const httpHeaders = {
+      'Accept': 'application/json, text/plain, */*',
+      'Content-Type': 'application/json',
+      'X-Origin': '*'
+    };
+
+
+    // let input = JSON.stringify(JSON.parse(this.state.userInput));
```
Comment (ericliu121187) [inline frontend/src/components/Textarea.js]: remove comment and line break above this

**Pull request merged**